### ISSUE TITLE
Bitbucket Server sync issue when auto merging pull requests

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#1120]
When trying to auto merge an auto versioning pull request, Ontrack will:

* wait until the PR is marked as mergeable
* send a request to merge the PR

With Bitbcuket Server, when PR checks include a minimal amount of builds being successful, there is a period during which the PR is marked as mergeable but is not:

* the build has not started yet - the PR is marked as mergeable
* the build has started and the PR becomes not mergeable

Between these two events, there is a window of time when Ontrack will attempt to merge and fail.

```
Pr cannot be merged because  "{"errors":[{"context":null,"message":"Merging the pull request has been vetoed.","exceptionName":"com.atlassian.bitbucket.pull.PullRequestMergeVetoedException","conflicted":false,"vetoes":[{"summaryMessage":"Not all required builds are successful yet","detailedMessage":"You still need a minimum of one successful build before this pull request can be merged."}]}]}" 
...
org.springframework.web.client.RestTemplate.postForObject(RestTemplate.java:443)
at net.nemerosa.ontrack.extension.stash.client.BitbucketClientImpl.mergePR(BitbucketClientImpl.kt:151)
at net.nemerosa.ontrack.extension.stash.scm.BitbucketServerSCMExtension$BitbucketServerSCM.waitAndMerge(BitbucketServerSCMExtension.kt:186) 
```

**Comment by dcoraboeuf** [comment]
The check for the "mergeability" must include the vetoes (part of the response):

```json
{
  "outcome": "CLEAN",
  "vetoes": [
    {
      "summaryMessage": "Insufficient branch permissions",
      "detailedMessage": "You have insufficient permissions to rebase 'refs/heads/feature-branch'."
    }
  ],
  "conflicted": true
}
```
https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-pull-requests/#api-api-latest-projects-projectkey-repos-repositoryslug-pull-requests-pullrequestid-merge-get

**Comment by dcoraboeuf** [comment]
Available in 4.7.18
